core/src/Kokkos_Core.hpp

```diff
@@ -55,6 +55,7 @@
 #include <Kokkos_Half.hpp>
 #include <KokkosCore_Config_DeclareBackend.hpp>
 
+#include <Kokkos_ArithmeticTraits.hpp>
```

Comment:
Note that I called the header `<Kokkos_ArithmeticTraits.hpp>` because `<Kokkos_NumericTraits.hpp>` already exists but that I would like to reclaim the file name.